classes/services/PublicationService.inc.php

```diff
@@ -191,6 +192,31 @@ public function validatePublishPublication($hookName, $args) {
 		}
 	}
 
+	/**
+	 * Make validation checks against payments, if configured
+	 * @param $hookName string
+	 * @param $args array [
+	 *		@option array Validation errors already identified
+	 *		@option Publication The publication to validate
+	 *		@option Submission The submission of the publication being validated
+	 *		@option array The locales accepted for this object
+	 *		@option string The primary locale for this object
+	 * ]
+	 */
+	public function validatePublishPayments($hookName, $args) {
+		$errors =& $args[0];
+		$submission = $args[2];
+
+		$context = Services::get('context')->get($submission->getData('contextId'));
+		$paymentManager = \Application::getPaymentManager($context);
+		$completedPaymentDao = \DAORegistry::getDAO('OJSCompletedPaymentDAO'); /* @var $completedPaymentDao OJSCompletedPaymentDAO */
+		$publicationFeeEnabled = $paymentManager->publicationEnabled();
+		$publicationFeePayment = $completedPaymentDao->getByAssoc(null, PAYMENT_TYPE_PUBLICATION, $submission->getId());
+		if ($publicationFeeEnabled && !$publicationFeePayment) {
+			$errors['issueId'] = __('editor.article.payment.publicationFeeNotPaid');
+		}
+	}
+
```

Comment:
I'd recommend rolling this into the `vlaidatePublishPublication` method. Typically I've been putting all validation checks in one place where possible.